### PR TITLE
fix: show nft amount '0'

### PR DESCRIPTION
**Pull request by lelenei**
closes #

### CODE REVIEW

File: packages/maskbook/src/components/InjectedComponents/NFT/NFTAvatar.tsx [hunk 1]
```diff
@@ -242,10 +242,18 @@ export async function getNFT(address: string, tokenId: string) {
...
+        symbol: order?.paymentTokenContract?.symbol ?? 'ETH',
         image: asset.imageUrl ?? asset.imagePreviewUrl ?? '',
     }
 }
+
+export function useNFT(address: string, tokenId: string) {
+    return useAsyncRetry(async () => {
+        const nft = await getNFT(address, tokenId)
+        return nft
+    }, [address, tokenId, getNFT])
```
Comment (UncleBill) [inline packages/maskbook/src/components/InjectedComponents/NFT/NFTAvatar.tsx:254]:
```suggestion
    }, [address, tokenId])
```

`getNFT` is a static function, no need to add to dependencies array.

File: packages/maskbook/src/components/InjectedComponents/NFT/NFTBadge.tsx [hunk 2]
```diff
@@ -50,18 +49,21 @@ const useStyles = makeStyles()({
...
-    useAsync(async () => {
-        const { amount, symbol } = await getNFT(avatar.address, avatar.tokenId)
+    const { value = { amount: '0', symbol: 'ETH' }, loading } = useNFT(avatar.address, avatar.tokenId)
+
+    const { amount, symbol } = value
+    useEffect(() => {
         setAmount_(amount)
         setSymbol_(symbol)
-    }, [avatar])
+    }, [amount, symbol])
```
Comment (UncleBill) [inline packages/maskbook/src/components/InjectedComponents/NFT/NFTBadge.tsx:66]: Break into two `useEffect`s

```ts
useEffect(() => {
        setAmount_(amount)
    }, [amount])
useEffect(() => {
        setSymbol_(symbol)
    }, [symbol])
```

Or, looks like you don't need `amount_` and `symbol_` anymore, just use `amount` and `symbol` instead direactly.

File: packages/maskbook/src/components/InjectedComponents/NFT/NFTAvatar.tsx [hunk 3]
```diff
@@ -224,6 +224,27 @@ export interface AvatarMetaDB {
...
+    symbol: string
+    name: string
+    image: string
+}
+
+const NFTAvatarCache = new Map<string, AvatarMetaDB>()
+const NFTCache = new Map<string, NFT>()
+
+function getNFTCache(userId: string) {
+    if (NFTCache.has(userId)) {
```
Comment (guanbinrui) [inline packages/maskbook/src/components/InjectedComponents/NFT/NFTAvatar.tsx:238]:
```suggestion
    return NFTCache.get(userId)
```

File: packages/maskbook/src/components/InjectedComponents/NFT/NFTAvatar.tsx [hunk 4]
```diff
@@ -242,10 +263,21 @@ export async function getNFT(address: string, tokenId: string) {
...
     }
 }
+
+export function useNFT(userId: string, address: string, tokenId: string) {
+    return useAsyncRetry(async () => {
+        let nft = getNFTCache(userId)
+        if (nft) return nft
+        nft = await getNFT(address, tokenId)
+        setNFTCache(userId, nft)
+        return nft
```
Comment (guanbinrui) [inline packages/maskbook/src/components/InjectedComponents/NFT/NFTAvatar.tsx:277]:
```suggestion
        setNFTCache(userId, getNFTCache(userId) ?? await getNFT(address, tokenId))
        return getNFTCache(userId)
```

File: packages/maskbook/src/components/InjectedComponents/NFT/NFTAvatar.tsx (hunk 3 above)
Comment (septs) [reply packages/maskbook/src/components/InjectedComponents/NFT/NFTAvatar.tsx:238]: why not use `lodash.memoize`?

see https://lodash.com/docs/4.17.15#memoize

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**